src/Algebra/GCD.agda

```diff
@@ -0,0 +1,62 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of irreducibility, primality, coprimality, GCD.
+------------------------------------------------------------------------
+
+-- You're unlikely to want to use this module directly. Instead you
+-- probably want to be importing the appropriate module from
+-- `Algebra.Properties.CancellativeCommutativeSemiring`.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra.Core using (Op₂)
+import Algebra.Divisibility
+open import Data.Product using (_×_)
+open import Data.Sum using (_⊎_)
+open import Function using (flip)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; Symmetric)
+open import Relation.Nullary using (¬_)
+open import Relation.Unary using (Pred)
+
+module Algebra.GCD
+  {β ℓ} {A : Set β} (_≈_ : Rel A ℓ) (0# 1# : A) (_*_ : Op₂ A)
+  where
+
+open import Algebra.Divisibility _≈_ _*_
+
+------------------------------------------------------------------------------
+-- Definitions
+
+infix 4 _≉_
+
+_≉_ : Rel A _
+x ≉ y = ¬ (x ≈ y)
+
+IsIrreducible : Pred A (β ⊔ ℓ)
```

Comment:
I'm not sure the definitions of `IsIrreducible`, `IsPrime`, and `Coprime` belong in the `GCD` module. I'd move them to a separate `Algebra.Primality` module.

Reply:
Moved to Algebra.Primality.